In [1]:
import pandas as pd

In [2]:
# Set display options for better viewing
pd.set_option('display.max_columns', None)
pd.set_option('display.width', 200)

In [3]:
# Load the data
df = pd.read_csv('../data/netflix_shows_complete.csv')

In [4]:
# How many rows and columns?
print(f"Dataset shape: {df.shape}")
print(f"We have {df.shape[0]} shows and {df.shape[1]} features")

Dataset shape: (1500, 23)
We have 1500 shows and 23 features


In [5]:
# What columns do we have?
print(df.columns.tolist())

['id', 'name', 'first_air_date', 'popularity', 'vote_average', 'vote_count', 'status', 'in_production', 'num_seasons', 'num_episodes', 'genres', 'type', 'original_language', 'origin_country', 'avg_episode_runtime', 'show_age_days', 'days_since_last_episode', 'keywords', 'last_air_date', 'us_content_rating', 'imdb_id', 'created_by', 'homepage']


In [6]:
# Last few rows
df.tail()

,id,name,first_air_date,popularity,vote_average,vote_count,status,in_production,num_seasons,num_episodes,genres,type,original_language,origin_country,avg_episode_runtime,show_age_days,days_since_last_episode,keywords,last_air_date,us_content_rating,imdb_id,created_by,homepage
1495,308014,Berlin and the Lady with an Ermine,2026-05-15,1.7502,0.000,0,In Production,True,1,8,"Drama, Crime",Miniseries,es,ES,NaN,-109.0,NaN,"heist, miniseries, art heist",NaN,NaN,NaN,"Álex Pina, Esther Martínez Lobato",https://www.netflix.com/title/82152349
1496,224941,The Boroughs,NaN,1.7498,0.000,0,In Production,True,1,1,"Drama, Sci-Fi & Fantasy, Mystery",Scripted,en,US,NaN,NaN,NaN,"supernatural, retirement community",NaN,NaN,tt27557289,"Jeffrey Addiss, Will Matthews",NaN
1497,122672,The Sons of Sam: A Descent into Darkness,2021-05-05,1.7491,6.961,129,Ended,False,1,4,"Documentary, Crime",Miniseries,en,US,NaN,1727.0,1727.0,"new york city, 1970s, cult, serial killer, tru...",2021-05-05,TV-MA,tt13156316,NaN,https://www.netflix.com/title/81059887
1498,93878,Criminal: Spain,2019-09-20,1.7488,6.571,98,Ended,False,1,3,"Crime, Drama",Miniseries,es,ES,40.0,2320.0,2320.0,"detective, investigation, miniseries, criminal...",2019-09-20,TV-MA,tt10986050,"Jim Field Smith, George Kay",https://www.netflix.com/title/81020523
1499,256052,Inside,2024-06-02,1.7480,6.375,8,Returning Series,True,2,15,Reality,Reality,en,GB,NaN,603.0,308.0,NaN,2025-03-24,NaN,tt32510002,NaN,https://www.youtube.com/playlist?list=PLjkZIuJ...


In [7]:
# Data types
df.dtypes

id                           int64
name                        object
first_air_date              object
popularity                 float64
vote_average               float64
vote_count                   int64
status                      object
in_production                 bool
num_seasons                  int64
num_episodes                 int64
genres                      object
type                        object
original_language           object
origin_country              object
avg_episode_runtime        float64
show_age_days              float64
days_since_last_episode    float64
keywords                    object
last_air_date               object
us_content_rating           object
imdb_id                     object
created_by                  object
homepage                    object
dtype: object

In [8]:
# Summary statistics
df.describe()

,id,popularity,vote_average,vote_count,num_seasons,num_episodes,avg_episode_runtime,show_age_days,days_since_last_episode
count,1500.000000,1500.000000,1500.000000,1500.000000,1500.000000,1500.000000,756.000000,1492.000000,1485.000000
mean,157191.126667,6.441002,7.061751,329.362667,1.950000,23.702667,45.282489,1644.855228,1225.305724
std,75759.593456,13.743460,1.159450,1208.769906,2.535045,109.490510,58.760052,1348.478030,908.330675
min,502.000000,1.558000,0.000000,0.000000,1.000000,1.000000,4.000000,-109.000000,3.000000
25%,90283.750000,2.463375,6.609750,25.000000,1.000000,7.000000,25.000000,717.750000,424.000000
50%,131538.500000,3.617200,7.200000,66.000000,1.000000,10.000000,40.500000,1439.000000,1118.000000
75%,227013.500000,6.476925,7.700000,197.000000,2.000000,20.000000,50.000000,2324.000000,1928.000000
max,310070.000000,427.979800,10.000000,20437.000000,56.000000,3161.000000,1100.000000,20531.000000,4685.000000


In [9]:
# Missing values
missing = df.isnull().sum()
missing[missing>0]

first_air_date               8
genres                       5
origin_country               1
avg_episode_runtime        744
show_age_days                8
days_since_last_episode     15
keywords                   128
last_air_date               15
us_content_rating          214
imdb_id                     23
created_by                 467
homepage                    26
dtype: int64

In [10]:
# Status distribution
print(df['status'].value_counts())

status
Ended               914
Returning Series    357
Canceled            214
In Production        15
Name: count, dtype: int64


#### Initial Insights
- 214 shows have been cancelled, and 357 shows are returning.
- 14.27% of shows have been canceled.
- Columns with missing data:
first_air_date               8
genres                       5
origin_country               1
avg_episode_runtime        744
show_age_days                8
days_since_last_episode     15
keywords                   128
last_air_date               15
us_content_rating          214
imdb_id                     23
created_by                 467
homepage                    26
- Minimum popularity rating: 1.56
- Maximum popularity rating: 427.98
